sql/sql.go

```diff
@@ -0,0 +1,250 @@
+// Package sql provides an way to create sql.DB objects out of an RQLite store.
+//
+// The goal of this package is to provide an interface that looks and feels
+// identical to a plain old sql.DB one. The key here is really the
+// RqliteSQLWrapper interface: users create a store, and provide two functions
+// which execute queries on the leader (using whatever transport API they have
+// defined), and users can use this Store like a regular sql.DB
+package sql
+
+import (
+	"database/sql"
+	"database/sql/driver"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/raft"
+
+	"github.com/rqlite/rqlite/db"
+	"github.com/rqlite/rqlite/store"
+)
+
+// RqliteSQLWrapper is the user-supplied interface that allows the driver to
+// resolve the rqlite store and communicate with its members.
+type RqliteSQLWrapper interface {
+	// Store returns the local store that this wrapper wraps.
+	Store() (*store.Store, error)
+
+	// ExecOn should execute the specified query on the specified address
```

Comment:
If the current store is not the leader, you need some way to talk to the leader to actually execute the query; this is the way to provide that. This way, on every node that opens one of these, every query succeeds, instead of sometimes failing with ErrNotLeader and leaving you to handle it.

I can draft an implementation that's rqlite specific if you like to demo it.